### PR TITLE
Add check for usage of secured user id

### PR DESCRIPTION
**Pull request by steventir**
please review

### CODE REVIEW

File: main/resources/reco_script.js [hunk 1]
```diff
@@ -164,20 +164,20 @@ document.addEventListener('DOMContentLoaded', function () {
         	addURLParam("_K14_", formatWith("'", params['data-k14']));
 
 		var url;
-		if (!params['data-h']) { // non-secure id
-			url = serviceURL + addURLParam("UserId", formatWith("'", params['data-userid']));
-		} else {
+		var headers = {};
+		if (params['data-h'] && params['data-userid']) {
```
Comment (mouradsap) [inline main/resources/reco_script.js]: We have a requirement that if the Recommendation Scenario is not marked with _**Secure User = true**_, then it should still be possible to pass a user in the header without a hash. We should be able to support that. We should be able to obtain this information from the L54 string.

File: main/resources/reco_script.js [hunk 2]
```diff
@@ -165,10 +165,21 @@ document.addEventListener('DOMContentLoaded', function () {
...
 		var headers = {};
-		if (params['data-h'] && params['data-userid']) {
+		var l54 = parseL54(params["data-l54"]);
+
+		if (l54.secure_user) {
 			url = serviceURL;
-			headers['_u_'] = params['data-userid'];
-			headers['_h_'] = params['data-h'];
+			if (params['data-userid']) {
+				headers['_u_'] = params['data-userid'];
```
Comment (mouradsap) [inline main/resources/reco_script.js:173]: It should not be necessary to override the header field userId with the userId from the URL.
If the scenario has secure_id = true, take the userId from _u_, and ignore the userId from the URL
If the scenario has secure_id = false, then accept the user Id from the URL or from the _u_ (if both are provided we can prioritize the one in the URL param.

File: main/resources/reco_script.js [hunk 3]
```diff
@@ -186,7 +197,21 @@ document.addEventListener('DOMContentLoaded', function () {
 				return data;
 			}
     	});
-    }
+	}
+
+	function parseL54(b64) {
```
Comment (mouradsap) [inline main/resources/reco_script.js:202]: The method name should be more decodeL54, we're not only parsing :)
The method looks alright, we just need to test it well

File: main/resources/reco_script.js [hunk 4]
```diff
@@ -165,10 +165,21 @@ document.addEventListener('DOMContentLoaded', function () {
...
+				console.log("WARNING: missing userid parameter");
+			}
+
+			if (params['data-h']) {
+				headers['_h_'] = params['data-h'];
+			} else {
+				console.log("WARNING: missing h parameter");
+			}
 		} else {
 			url = serviceURL + addURLParam("UserId", formatWith("'", params['data-userid']));
```
Comment (mouradsap) [inline main/resources/reco_script.js:184]: So to confirm, when the scenario has secure_user=false, we should still be able to consume the userId from _u_ without a hash. If the userId is already provided in the params['data-userid'], we can prioritize that one and use it.

File: main/resources/reco_script.js [hunk 5]
```diff
@@ -165,10 +165,21 @@ document.addEventListener('DOMContentLoaded', function () {
...
-			headers['_u_'] = params['data-userid'];
-			headers['_h_'] = params['data-h'];
+			if (params['data-userid']) {
+				headers['_u_'] = params['data-userid'];
+			} else {
+				console.log("WARNING: missing userid parameter");
+			}
+
+			if (params['data-h']) {
+				headers['_h_'] = params['data-h'];
```
Comment (mouradsap) [inline main/resources/reco_script.js:179]: If secure_user = true, then both _u_ and _h_ have to be provided together, if the _h_ is missing, we ignore the _u_

File: main/resources/reco_script.js (hunk 3 above)
Comment (tirsap) [reply main/resources/reco_script.js:202]: nice catch, will change!

**Pull request merged**